src/report.rs

```diff
@@ -173,6 +180,50 @@ impl<P: Package, VS: VersionSet> fmt::Display for External<P, VS> {
     }
 }
 
+/// Trait for formatting outputs in the reporter.
+pub trait ReportFormatter<P: Package, VS: VersionSet> {
+    /// Output type of the report.
+    type Output;
```

Comment:
The current code only supports `Output = String`, eventually I prefer these functions return `impl Display` or take a `&mut Write`. Do you think `Output` pulls its weight? We could have the methods return `String` for now and fix it in a follow-up, or have the methods take `&mut Write` and return the bubbled up error. Thoughts?